robot_smach_states/src/robot_smach_states/util/designators/utility.py

```diff
@@ -8,10 +8,28 @@
 import operator
 
 # TU/e Robotics
-from .core import Designator, VariableDesignator
+from robot_smach_states.util.designators.core import Designator, VariableDesignator
 
 __author__ = 'loy'
 
+def resolve(designator_or_value):
```

Comment:
Beautiful! After this PR we should look where we should use this in robot_smach_states.

Reply:
I don't agree with this. This is only needed in case of bad management of your variables. This allows for creating a bigger mess, while we should reduce the number of possible variable types.

Reply:
I disagree. This designator or value stuff already happens a lot. Mostly for things that might be but don't have to be defined at construction time. For example with area names

Reply:
Jep, but that doesn't mean we should continue in the wrong direction. In case that the value is already known on construction you could put the value in a designator, therefore you always have a designator and never a value.

Reply:
I still disagree. But I think this is a discussion to have on a tuesday evening.

Reply:
You both have a point here. 
This `resolve` function allows to more easily upgrade a parameter to be a Designator later on, but otherwise I agree it should not be necessary.

Reply:
The difficult thing here is that oftentimes, you only find out something should have been runtime variable at a later time. To not change the external interface, you modify the internals to accept both a concrete value *and* Designator. This is a pattern I've seen a few times at least. 
So, reusing the code for that pattern is handy. 

The other option is to make larger changes to a constructor and all it's uses each time we find that a parameter should actually be runtime variable. But this is error-prone...

Reply:
I've updated this PR to match my advice at the bottom of my comment. 

The ease is nice, but clarity is more important.

Reply:
I still don't like the function. This way you are facilitating the bad habits. I don't say we should change the old code, but we are now aware of the pattern. So we should be forced to think more about the design, before we start coding. Adding this function doesn't force us in that direction.

Reply:
I'll remove the function as well, gimme a sec.

Reply:
1800 seconds have passed Loy. Please remove this function before merging